File: testdata/polyglot/go/webgaps.go
// Deliberately vulnerable appsec fixture. Every issue is planted. DO NOT fix.
// Web-gap classes caught by argus/curated rules the registry packs miss.
package webgaps

import (
	"crypto/tls"
	"net/http"
	"os"
	"path/filepath"
)

func ssrf(userURL string) (*http.Response, error) {
	// PLANT(go-ssrf-web, min-profile=standard, CWE-918): request to a variable URL (argus/curated)
	return http.Get(userURL)
}

func read(base, userInput string) ([]byte, error) {
	// PLANT(go-path-web, min-profile=standard, CWE-22): path joined from unsanitized input (argus/curated)
	return os.ReadFile(filepath.Join(base, userInput))
}

func tlsConfig() *tls.Config {
	// PLANT(go-tls-skip-verify, min-profile=standard, CWE-295): certificate verification disabled (argus/curated)
	return &tls.Config{InsecureSkipVerify: true}
}

File: testdata/polyglot/js/webgaps.js
// Deliberately vulnerable appsec fixture. Every issue is planted. DO NOT fix.
// Web-gap classes caught by argus/curated rules the registry packs miss.
const axios = require("axios");
const fs = require("fs");

app.get("/proxy", (req, res) => {
  // PLANT(js-ssrf-web, min-profile=standard, CWE-918): request to a request-controlled URL (argus/curated)
  return axios.get(req.query.url);
});

app.get("/file", (req, res) => {
  // PLANT(js-path-web, min-profile=standard, CWE-22): fs read from unsanitized request input (argus/curated)
  return fs.readFileSync(req.query.file);
});

app.get("/go", (req, res) => {
  // PLANT(js-open-redirect, min-profile=standard, CWE-601): redirect to a request-controlled target (argus/curated)
  res.redirect(req.query.next);
});
